dot/core/helpers.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package core
+
+func concatenateByteSlices(slices [][]byte) (concatenated []byte) {
+	length := 0
+	for i := range slices {
+		length += len(slices[i])
+	}
+	concatenated = make([]byte, 0, length)
+	for _, slice := range slices {
+		concatenated = append(concatenated, slice...)
+	}
+	return concatenated
+}
```

Comment:
would suggest to use [bytes.Join](https://pkg.go.dev/bytes#Join)
```go
s := [][]byte{[]byte("foo"), []byte("bar"), []byte("baz")}
bytes.Join(s, nil)
```

```suggestion
```

Reply:
Nice I didn't know about that one!! 👍

Reply:
Nice!